pkg/apis/options/load.go

```diff
@@ -54,6 +54,72 @@ func Load(configFileName string, flagSet *pflag.FlagSet, into interface{}) error
 	return nil
 }
 
+// LoadYAML will load a YAML based configuration file into the options interface provided.
+func LoadYAML(configFileName string, opts interface{}) error {
+	buffer, err := loadAndSubstituteEnvs(configFileName)
+	if err != nil {
+		return err
+	}
+
+	// Generic interface for loading arbitrary yaml structure
+	var intermediate map[string]interface{}
+
+	if err := yaml.Unmarshal(buffer, &intermediate); err != nil {
+		return fmt.Errorf("error unmarshalling config: %w", err)
+	}
+
+	return Decode(intermediate, opts)
+}
+
+func Decode(input interface{}, result interface{}) error {
+	// Using mapstructure to decode arbitrary yaml structure into options and
+	// merge with existing values instead of overwriting everything. This is especially
+	// important as we have a lot of default values for boolean which are supposed to be
+	// true by default. Normally by just parsing through yaml all booleans that aren't
+	// referenced in the config file would be parsed as false and we cannot identify after
+	// the fact if they have been explicitly set to false or have not been referenced.
+	decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
+		DecodeHook:           mapstructure.ComposeDecodeHookFunc(toDurationHookFunc()),
+		Metadata:             nil,    // Don't track any metadata
+		Result:               result, // Decode the result into the prefilled options
+		TagName:              "json", // Parse all fields that use the yaml tag
+		ZeroFields:           false,  // Don't clean the default values from the result map (options)
+		ErrorUnused:          true,   // Throw an error if keys have been used that aren't mapped to any struct fields
+		IgnoreUntaggedFields: true,   // Ignore fields in structures that aren't tagged with yaml
+	})
+
+	if err != nil {
```

Comment:
Prefer to delete this new line to keep the error tied to the logic that produces it